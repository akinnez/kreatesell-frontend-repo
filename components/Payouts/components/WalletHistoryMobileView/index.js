import { memo } from 'react'
import NoData from 'components/NoData'
import HistoryTag from '../HistoryTag'
import formatAccountNumber from '../../utils/formatAccountNumber'
import dateFormat from 'utils/dateFormat'
import styles from './index.module.scss'

const WalletHistoryMobileView = ({ histories }) => (
  <>
    {histories.length === 0 ? (
      <NoData />
    ) : (
      <ul className={styles.histories}>
        {histories?.map((history) => (
          <li key={history.id} className={styles.history}>
            <div className={styles.history__header}>
              <HistoryTag status={history.status} />
              <div className={styles['history__withdrawal-date']}>
                <span>Withdrawal Date</span>
                <p>{dateFormat(history.withdrawal_date)}</p>
              </div>
            </div>
            <ol>
              <li className={styles.history__detail}>
                <strong>Amount Withdrawn</strong>
                <span>
                  {history.currency} {history.amount}
                </span>
              </li>
              <li className={styles.history__detail}>
                <strong>Description</strong>
                <span>
                  {history.bank_name}{' '}
                  {`(${formatAccountNumber(history.bank_account)})`}
                </span>
              </li>
            </ol>
          </li>
        ))}
      </ul>
    )}
  </>
)

export default memo(WalletHistoryMobileView)
