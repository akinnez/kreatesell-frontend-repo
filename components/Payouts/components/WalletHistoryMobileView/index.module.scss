@use 'public/css/colors';
@use 'public/css/mixins';
@use 'public/css/screens';

.histories {
	display: grid;
	grid-template-columns: repeat(auto-fit, minmax(260px, 430px));
	justify-content: space-evenly;
	gap: 16px 20px;
}

.history {
	padding: 32px;
	background-color: colors.$base-white;
	border-radius: 8px;
}

.history__header {
	margin-bottom: 15px;
	@include mixins.flex(row, space-between, flex-start);
	gap: 15px 10px;

	> span {
		line-height: 1;
	}
}

.history__withdrawal-date {
	text-align: right;

	span {
		display: inline-block;
		margin-bottom: 5px;
		padding: 4px 8px;
		border-radius: 8px;
		font-size: 12px;
		color: colors.$base-blue;
		background-color: colors.$blue-100;
	}

	p {
		margin-bottom: 0;
		font-size: 14px;
		color: colors.$base-gray;
		line-height: 1;
	}
}

.history__detail {
	padding: 6px 10px;
	@include mixins.flex(row, flex-start, center);
	flex-wrap: wrap;
	gap: 2px 25px;

	&:nth-child(odd) {
		border-radius: 8px;
		background-color: colors.$white-200;
	}

	&:not(:last-of-type) {
		margin-bottom: 10px;
	}

	strong {
		flex: 0 0 135px;
		font-size: 14px;
		color: colors.$black-100;
	}

	span {
		font-size: 14px;
		color: colors.$base-gray;
	}
}

@include screens.screen('medium') {
	.history {
		padding: 32px 10px;
	}

	.history__header {
		flex-wrap: wrap;
	}

	.history__withdrawal-date {
		text-align: left;
	}

	.history__detail {
		padding-top: 4px;
		padding-bottom: 4px;
		column-gap: 10px;
	}
}
@include screens.screen('tablet') {
	.history__detail {
		flex-wrap: nowrap;
		& > strong {
			flex: 2;
		}
		& > span {
			flex: 5;
		}
	}
}
