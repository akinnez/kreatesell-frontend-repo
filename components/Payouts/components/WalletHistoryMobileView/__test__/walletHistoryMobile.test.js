import WalletHistoryMobileView from "..";
import renderer from "react-test-renderer";

const mockProps = {
  histories: [
    {
      id: 1,
      status: "Pending",
      withdrawal_date: "20/12/2022",
      currency: "NGN",
      amount: 1000,
      bank_name: "UBA",
      bank_account: "00093999393",
    },
    {
      id: 2,
      status: "Successful",
      withdrawal_date: "21/12/2022",
      currency: "USD",
      amount: 1000,
      bank_name: "BANK OF AMERICA",
      bank_account: "00393999393",
    },
    {
      id: 3,
      status: "Failed",
      withdrawal_date: "10/12/2022",
      currency: "GBP",
      amount: 1000,
      bank_name: "UBA",
      bank_account: "00123999393",
    },
  ],
};

describe("WalletHistoryMobileView : ", () => {
  it("renders a snapshot of <WalletHistoryMobileView /> when no payouts is supplied ", () => {
    const walletHistoryView = renderer
      .create(<WalletHistoryMobileView histories={[]} />)
      .toJSON();
    expect(walletHistoryView).toMatchSnapshot();
  });
  it("renders a snapshot of <WalletHistoryMobileView /> when payouts data is supplied ", () => {
    const walletHistoryView = renderer
      .create(<WalletHistoryMobileView {...mockProps} />)
      .toJSON();
    expect(walletHistoryView).toMatchSnapshot();
  });
});
