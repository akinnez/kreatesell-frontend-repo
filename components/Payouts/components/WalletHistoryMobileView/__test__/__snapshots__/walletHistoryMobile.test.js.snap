// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletHistoryMobileView :  renders a snapshot of <WalletHistoryMobileView /> when no payouts is supplied  1`] = `
<div
  className="no__data"
>
  <img
    src="[jest.find('../utils')]"
  />
  <p>
    No record yet
  </p>
</div>
`;

exports[`WalletHistoryMobileView :  renders a snapshot of <WalletHistoryMobileView /> when payouts data is supplied  1`] = `
<ul
  className="histories"
>
  <li
    className="history"
  >
    <div
      className="history__header"
    >
      <span
        className="tag orange"
      >
        Pending
      </span>
      <div
        className="history__withdrawal-date"
      >
        <span>
          Withdrawal Date
        </span>
        <p>
          undefined NaN NaN · NaN:NaN AM
        </p>
      </div>
    </div>
    <ol>
      <li
        className="history__detail"
      >
        <strong>
          Amount Withdrawn
        </strong>
        <span>
          NGN
           
          1000
        </span>
      </li>
      <li
        className="history__detail"
      >
        <strong>
          Description
        </strong>
        <span>
          UBA
           
          (*******9393)
        </span>
      </li>
    </ol>
  </li>
  <li
    className="history"
  >
    <div
      className="history__header"
    >
      <span
        className="tag green"
      >
        Successful
      </span>
      <div
        className="history__withdrawal-date"
      >
        <span>
          Withdrawal Date
        </span>
        <p>
          undefined NaN NaN · NaN:NaN AM
        </p>
      </div>
    </div>
    <ol>
      <li
        className="history__detail"
      >
        <strong>
          Amount Withdrawn
        </strong>
        <span>
          USD
           
          1000
        </span>
      </li>
      <li
        className="history__detail"
      >
        <strong>
          Description
        </strong>
        <span>
          BANK OF AMERICA
           
          (*******9393)
        </span>
      </li>
    </ol>
  </li>
  <li
    className="history"
  >
    <div
      className="history__header"
    >
      <span
        className="tag red"
      >
        Failed
      </span>
      <div
        className="history__withdrawal-date"
      >
        <span>
          Withdrawal Date
        </span>
        <p>
          Oct 12 2022 · 0:00 AM
        </p>
      </div>
    </div>
    <ol>
      <li
        className="history__detail"
      >
        <strong>
          Amount Withdrawn
        </strong>
        <span>
          GBP
           
          1000
        </span>
      </li>
      <li
        className="history__detail"
      >
        <strong>
          Description
        </strong>
        <span>
          UBA
           
          (*******9393)
        </span>
      </li>
    </ol>
  </li>
</ul>
`;
